### PR TITLE
Implement meta2-mover

### PR DESCRIPTION
**Pull request by AymericDu**

### CODE REVIEW

File: bin/oio-meta2-mover [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library.
+
+import argparse
+
+from oio.directory.meta1 import Meta1RefMapping
+
+def make_arg_parser():
+    descr = "Move a base of source service to destination service"
```
Comment (vincent-legoll) [inline bin/oio-meta2-mover]: s/of/from/ non ?

File: oio/directory/meta.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+            self.logger.info("old: %s, new: %s", old_peers, new_peers)
+            service_type = self._get_service_type_by_base(base)
+            cid, _ = self.get_cid_and_seq(base)
+            try:
+                self.admin.set_peers(service_type, cid=cid, peers=peers)
+                all_peers_ok = True
+                for svc_to in new_peers:
+                    this_peer_ok = False
+                    for svc_from in old_peers:
+                        self.logger.info("Copying base %s from %s to %s",
```
Comment (vincent-legoll) [inline oio/directory/meta.py]: This one is OK ;-)

File: oio/directory/meta1.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+                service = raw_services.pop(src_service)
+                raw_services[dest_service] = service
+                valid_bases.add(base)
+                self.moved_bases.add(base)
+                self.raw_services_by_base[base] = old_peers
+                self.services_by_base[base] = raw_services
+                self.service_type_by_base[base] = service_type
+                self.args_by_base[base] = service['args']
+
+        if not valid_bases:
```
Comment (vincent-legoll) [inline oio/directory/meta1.py]: Couldn't you get more precise error reporting ?
"X or Y is wrong" is not super helpful as an error message, or maybe there's another way to know which one is wrong...

File: oio/directory/admin.py [hunk 4]
```diff
@@ -59,10 +59,12 @@ def election_debug(self, params, **kwargs):
         return body
 
     @loc_params
-    def election_leave(self, params, **kwargs):
+    def election_leave(self, params, id=None, **kwargs):
```
Comment (fvennetier) [inline oio/directory/admin.py]: `id` is a reserved keyword. Besides it's not clear what this ID is. Is it a service ID?

**Pull request merged**